src/main/java/org/broadinstitute/hellbender/tools/spark/sv/StructuralVariationDiscoveryPipelineSpark.java

```diff
@@ -0,0 +1,177 @@
+            return ctx.parallelize( viaSAMRoute(alignedAssemblyOrExcuseList, header, toolLogger) );
+        }
+
+        @VisibleForTesting
+        public List<AlignedAssembly.AlignedContig> viaSAMRoute(final List<AlignedAssemblyOrExcuse> alignedAssemblyOrExcuseList,
+                                                               final SAMFileHeader header, final Logger toolLogger) {
+
+            final SAMFileHeader cleanHeader = new SAMFileHeader(header.getSequenceDictionary());
+            final List<AlignedAssembly.AlignedContig> parsedAlignments
+                    = AlignedAssemblyOrExcuse.turnIntoSAMRecordsForEachAssembly(cleanHeader, alignedAssemblyOrExcuseList).stream()
+                    .filter(samRecords -> Utils.stream(samRecords).anyMatch(sam -> !sam.getReadUnmappedFlag()))
+                    .map(iterable -> Utils.stream(iterable).filter(sam -> !sam.getNotPrimaryAlignmentFlag()).collect(Collectors.toList()))
+                    .filter(iterable -> !IterableUtils.isEmpty(iterable))
+                    .map(iterable -> DiscoverVariantsFromContigAlignmentsSAMSpark.SAMFormattedContigAlignmentParser.parseReadsAndBreakGaps(Utils.stream(iterable).map(SAMRecordToGATKReadAdapter::new).collect(Collectors.toList()), header, SVConstants.DiscoveryStepConstants.GAPPED_ALIGNMENT_BREAK_DEFAULT_SENSITIVITY, toolLogger))
+                    .collect(Collectors.toList());
+
+            return parsedAlignments.isEmpty() ? Collections.emptyList() : parsedAlignments;
+        }
+
+        /**
+         * Filter out "failed" assemblies and turn the alignments of contigs into custom {@link AlignedAssembly.AlignmentInterval} format.
+         */
+        @VisibleForTesting
+        public static List<AlignedAssembly.AlignedContig> directToAlignmentInterval(final Iterable<AlignedAssemblyOrExcuse> alignedAssemblyOrExcuseIterable,
+                                                                                    final List<String> refNames, final SAMFileHeader header) {
+
+            return Utils.stream(alignedAssemblyOrExcuseIterable)
+                    .filter(alignedAssemblyOrExcuse -> alignedAssemblyOrExcuse.getErrorMessage() == null)
+                    .map(alignedAssembly -> forEachAssemblyNotExcuse(alignedAssembly, refNames, header))
+                    .flatMap(Utils::stream)                                     // size == total # of contigs' from all successful assemblies
+                    .filter(contig -> !contig.alignmentIntervals.isEmpty())     // filter out unmapped and contigs without primary alignments
+                    .collect(Collectors.toList());
+        }
+
+        /**
+         * Work on "successful" assembly and turn its contigs' alignments to custom {@link AlignedAssembly.AlignmentInterval} format.
+         */
+        @VisibleForTesting
+        public static Iterable<AlignedAssembly.AlignedContig> forEachAssemblyNotExcuse(final AlignedAssemblyOrExcuse alignedAssembly,
```

Comment:
Please change the names of these methods to describe what they do (`convertToAlignmentIntervals`)